### ISSUE TITLE
Dead jobs callback/notification

### ISSUE THREAD

**Issue by jvenezia** [mperham/sidekiq#3531]
Hi,

I would like to get notifications (slack, mail, sentry or anything) when a job enters the dead queue.
An `on_dead` callback would do the job.
I could poll and check if there is dead jobs every x minutes but It's not ideal imo.

Do we have that? What would be the best way to achieve that?

Thanks!

**Comment by mperham** [comment]
Best approach would be to add this:

```ruby
Sidekiq.configure_server do |config|
  config.on(:dead) do |job_hash|
    # do something, e.g.
    # SlackNotifier.dead_job(job)
  end
end
```

**Comment by jvenezia** [comment]
That's what I'm looking for :)

Thank you!

**Comment by jvenezia** [comment]
Is this mentioned somewhere on the Wiki? Can't find it.

**Comment by mperham** [comment]
It's not implemented but would take about 10 lines of code. I look forward to your PR!  :-)

**Comment by jvenezia** [comment]
![facepalm](https://user-images.githubusercontent.com/3819893/27687416-0e108030-5cd7-11e7-84a4-3c9d0dec661f.jpg)

Well ok... I'll do that.

**Comment by jvenezia** [comment]
Reading the code I found a similar feature available since 4.1.1 :

```
Sidekiq.configure_server do |config|
  config.default_retries_exhausted = -> (job, ex) do
    Sidekiq.logger.info "#{job['class']} job is now dead"
  end
end
```

**Comment by Galaxy83** [comment]
If anyone got here late and nothing works, here is the right approach:
```
Sidekiq.configure_server do |config|
  config.death_handlers << ->(job, ex) do
    # do stuff
    # job keys: ["class", "args", "retry", "queue", "backtrace", "jid", "created_at", "enqueued_at", "error_message", "error_class", "failed_at", "retry_count", "error_backtrace"]
  end
end
```
